sled-agent/src/support_bundle/logs.rs

```diff
@@ -78,12 +80,53 @@ impl<'a> SupportBundleLogs<'a> {
     where
         Z: Into<String>,
     {
-        // We are using an M.2 device for temporary storage to assemble a zip
-        // file made up of all of the discovered zone's logs.
-        let current_internal_disks = self.internal_disks_rx.current();
-        let mut m2_debug_datasets = current_internal_disks.all_debug_datasets();
-        let tempdir = m2_debug_datasets.next().ok_or(Error::MissingStorage)?;
-        let mut tempfile = tempfile_in(tempdir)?;
+        // Attempt to find a U.2 device with the most available free space
+        // for temporary storage to assemble a zip file made up of all of the
+        // discovered zone's logs.
```

Comment:
Style nit I don't feel strongly about - I'd maybe move this chunk of code to a separate method? `dataset_for_temporary_storage()` or something

Reply:
Sorry I had saw the approval on mobile when I stepped away from the laptop and marked it for auto merge, I didn't catch this nit in time so here's the follow up: #8454